accounts-db/src/accounts_index/bucket_map_holder.rs

```diff
@@ -483,6 +577,211 @@ pub mod tests {
+        let bins = 1;
+        let config = AccountsIndexConfig {
+            index_limit: IndexLimit::Minimal,
+            ..Default::default()
+        };
+        let test = BucketMapHolder::<u64, u64>::new(bins, &config, 1);
+
+        assert!(test.should_flush_to_disk(0));
+        assert!(test.should_flush_to_disk(1000));
+        assert!(test.should_flush_to_disk(usize::MAX));
+    }
+
+    #[test]
+    fn test_in_mem_only_never_flushes() {
+        let bins = 1;
+        let config = AccountsIndexConfig {
+            index_limit: IndexLimit::InMemOnly,
+            ..Default::default()
+        };
+        let test = BucketMapHolder::<u64, u64>::new(bins, &config, 1);
+
+        assert!(!test.should_flush_to_disk(0));
+        assert!(!test.should_flush_to_disk(1000));
+        assert!(!test.should_flush_to_disk(usize::MAX));
+    }
+
+    #[test]
+    fn test_max_evictions_minimal() {
+        let bins = 1;
+        let config = AccountsIndexConfig {
+            index_limit: IndexLimit::Minimal,
+            ..Default::default()
+        };
+        let test = BucketMapHolder::<u64, u64>::new(bins, &config, 1);
+
+        assert_eq!(
+            test.max_evictions_for_threshold(0),
+            NonZeroUsize::new(1).unwrap()
+        );
+        assert_eq!(
+            test.max_evictions_for_threshold(1000),
+            NonZeroUsize::new(1000).unwrap()
+        );
+    }
+
+    #[test]
+    fn test_max_evictions_threshold() {
+        let bins = 1;
+        let threshold_entries = 1000;
+        let bytes_per_entry = InMemAccountsIndex::<u64, u64>::size_of_uninitialized()
+            + InMemAccountsIndex::<u64, u64>::size_of_single_entry();
+        let limit_bytes = (threshold_entries * bytes_per_entry) as u64;
+        let config = AccountsIndexConfig {
+            index_limit: IndexLimit::Threshold(limit_bytes),
+            ..Default::default()
+        };
+        let test = BucketMapHolder::<u64, u64>::new(bins, &config, 1);
+        let low_water_mark = test.threshold_entries_per_bin.unwrap().low_water_mark;
+        assert_eq!(low_water_mark, 313);
+        let expected_evictions = |current_entries: usize| {
+            NonZeroUsize::new(current_entries.saturating_sub(low_water_mark).max(1)).unwrap()
+        };
+
+        assert_eq!(
+            test.max_evictions_for_threshold(2000),
+            expected_evictions(2000)
+        );
+
+        assert_eq!(
+            test.max_evictions_for_threshold(1500),
+            expected_evictions(1500)
+        );
+
+        assert_eq!(
+            test.max_evictions_for_threshold(896),
+            expected_evictions(896)
+        );
+
+        assert_eq!(
+            test.max_evictions_for_threshold(500),
+            expected_evictions(500)
+        );
+        assert_eq!(
+            test.max_evictions_for_threshold(100),
+            expected_evictions(100)
+        );
+        assert_eq!(test.max_evictions_for_threshold(0), expected_evictions(0));
```

Comment:
What's the point of these different values being tested?

Reply:
done. Simplified the test. one value is enough.